### ISSUE TITLE
MAPE: nan RMSE: nan

### ISSUE THREAD

**Issue by SyedKumailHussainNaqvi** [unit8co/darts#2023]
i am trying to forecast the PV Power day ahead by using Darts Library models, but when i calculate the MAPE and RMSE error then its print MAPE: nan RMSE: nan
Here is my complete, kindly guide me why this is happening?

```python
from darts import TimeSeries
from darts.dataprocessing.transformers import Scaler
from darts.metrics import mape,mse
from darts.models import LinearRegressionModel, ExponentialSmoothing, ARIMA, RandomForest,Theta,RegressionEnsembleModel,NBEATSModel
from darts.utils.timeseries_generation import gaussian_timeseries, linear_timeseries, sine_timeseries
from darts.models import RNNModel, TCNModel, TransformerModel, NBEATSModel, BlockRNNModel

Target_Active_Power=TimeSeries.from_csv("/home/kumail/Energy_AI/PV/Dataset/0.5_year.csv", time_col="timestamp" , value_cols='Active_Power', freq="5T")
Input_Features =TimeSeries.from_csv("/home/kumail/Energy_AI/PV/Dataset/0.5_year.csv", time_col="timestamp" , value_cols=[
    'Current_Phase_Average',
    'Weather_Temperature_Celsius',
    'Weather_Relative_Humidity',
    'Global_Horizontal_Radiation',
    'Diffuse_Horizontal_Radiation',
    'Wind_Speed',
], freq="5T")
train_size = int(len(Target_Active_Power) * 0.8)
Target_train_series, Target_val_series = Target_Active_Power[:train_size], Target_Active_Power[train_size:]
Input_train_series, Input_val_series = Input_Features[:train_size], Input_Features[train_size:]
scaler_target = Scaler()
scaler_input = Scaler()
Train_Target_Active_Power_scaled = scaler_target.fit_transform(Target_train_series)
Train_Input_Features_scaled = scaler_input.fit_transform(Input_train_series)
val_Target_Active_Power_scaled = scaler_target.transform(Target_val_series)
val_Input_Features_scaled = scaler_input.transform(Input_val_series)
model_pastcov = BlockRNNModel(
    model="LSTM",
    input_chunk_length=288,
    output_chunk_length=10541,
    n_epochs=5,
    random_state=0,
    pl_trainer_kwargs={
      "accelerator": "gpu",
       "devices": 1
    },
)
model_pastcov.fit(
    series=Train_Target_Active_Power_scaled,
    past_covariates=Train_Input_Features_scaled,
    verbose=True,
);
pred_cov = model_pastcov.predict(n=10541, series=val_Target_Active_Power_scaled, past_covariates=val_Input_Features_scaled)
n_preds = scaler_target.inverse_transform(pred_cov)
mape_score = mape(n_preds,val_Target_Active_Power_scaled)
mse_score = mse(val_Target_Active_Power_scaled, n_preds)
rmse_score = mse_score ** 0.5
    
print(f'MAPE: {mape_score:.4f}')
print(f'RMSE: {rmse_score:.4f}')
```
output of this code
MAPE: nan
RMSE: nan

**Comment by dennisbader** [comment]
Hi @SyedKumailHussainNaqvi, this looks like there are some missing values either in your target or predicted series.

**Comment by SyedKumailHussainNaqvi** [comment]
Thank for your kind guidance, I fix this but i have another query about TransformerModel.
I have these train, validation and test series after scaling now i just predict the trained Transformer model on test series for forecasting without  n – The number of time steps after the end of the training time series, how its possible?

scaler = MinMaxScaler(feature_range=(-1, 1))
scaler_target = Scaler(scaler)
scaler_input = Scaler(scaler)
Train_Target_Active_Power_scaled = scaler_target.fit_transform(Target_train_series)
Train_Input_Features_scaled = scaler_input.fit_transform(Input_train_series)

val_Target_Active_Power_scaled = scaler_target.transform(Target_val_series)
val_Input_Features_scaled = scaler_input.transform(Input_val_series)

Test_Target_Active_Power_scaled = scaler_target.transform(Target_test_series)
Test_Input_Features_scaled = scaler_input.transform(Input_test_series)

my_model = TransformerModel(
    input_chunk_length=32,
    output_chunk_length=288,
    batch_size=128,
    n_epochs=1,
    #model_name="air_transformer",
    nr_epochs_val_period=100,
    d_model=32,
    nhead=16,
    num_encoder_layers=8,
    num_decoder_layers=8,
    dim_feedforward=256,
    dropout=0.1,
    activation="relu",
    #random_state=42,
    #save_checkpoints=True,
    force_reset=True,
    pl_trainer_kwargs={
      "accelerator": "gpu",
       "devices": 1
    },
)
my_model.fit(
    series=Train_Target_Active_Power_scaled,past_covariates=Train_Input_Features_scaled, val_series=val_Target_Active_Power_scaled,val_past_covariates=val_Input_Features_scaled,
    verbose=True,
);
pred_series = my_model.predict(n=len(Test_Target_Active_Power_scaled), series=Test_Target_Active_Power_scaled, past_covariates=Test_Input_Features_scaled)

Actual_Pred= scaler_target.inverse_transform(pred_series)
Actual_val_Target_Active_Power= scaler_target.inverse_transform(Test_Target_Active_Power_scaled)

mape_score = mape(Actual_Pred,Actual_val_Target_Active_Power)
mse_score = mse(Actual_Pred,Actual_val_Target_Active_Power)
mae_score = mae(Actual_Pred, Actual_val_Target_Active_Power)
print(f'MAPE: {mape_score:.4f}')
print(f'RMSE: {mse_score:.4f}')
print(f'MAE: {mae_score:.4f}')

**Comment by madtoinou** [comment]
Hi @SyedKumailHussainNaqvi,

> I have these train, validation and test series after scaling now i just predict the trained Transformer model on test series for forecasting without n – The number of time steps after the end of the training time series, how its possible?

I don't understand what you mean by this, your code snippet is clearly relying on `n` to indicate the length of the forecasts. This argument is mandatory.

If you want to forecast the series with the same index as your test series in order to compute metrics, you need to use `my_model.predict(n=len(test_series), series=validation_series, covariates=...)` as the prediction time index will start where the input `series` finishes.
